lib/rubocop/cop/performance/double_start_end_with.rb

```diff
@@ -0,0 +1,54 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop checks for double `#start_with?` or `#end_with?` calls
+      # separated by `||`. In some cases such calls can be replaced
+      # with an single `#start_with?`/`#end_with?` call.
+      #
+      # @example
+      #
+      #   @bad
+      #   str.start_with?("a") || str.start_with?(Some::CONST)
+      #   str.start_with?("a", "b") || str.start_with?("c")
+      #   var1 = ...
+      #   var2 = ...
+      #   str.end_with?(var1) || str.end_with?(var2)
+      #
+      #   @good
+      #   str.start_with?("a", Some::CONST)
+      #   str.start_with?("a", "b", "c")
+      #   var1 = ...
+      #   var2 = ...
+      #   str.end_with?(var1, var2)
+      class DoubleStartEndWith < Cop
+        MSG = 'Use `%{receiver}.%{method}(%{combined_args})` ' \
+              'instead of `%{original_code}`.'
+
+        def on_or(node)
+          receiver, method, args1, args2 = two_start_end_with_calls(node)
+          if receiver && args2.all?(&:pure?)
```

Comment:
I have to see what this function does, but I'm pretty sure it can't check whether a function is pure. :-)